test/resolver.test.ts

```diff
@@ -1,53 +1,55 @@
-import { Resolver } from 'did-resolver'
+import { Resolver } from 'did-resolver';
 import { getResolver } from '../src/index';
 import fetch from 'node-fetch';
+import { testDids } from './testDids';
 
-jest.setTimeout(10000)
+jest.setTimeout(10000);
 
-describe('EOSIO resolver', () => {
+describe('resolver tests', () => {
   const resolver = new Resolver(getResolver());
-
-  it('Resolve a complex EOS DID Document', async () => {
-    const eoscanadacomDid = 'did:eosio:eos:eoscanadacom';
-    console.log('Resolving ' + eoscanadacomDid);
-    const eosDidDocument = await resolver.resolve(eoscanadacomDid, { fetch });
-    console.log('DID Document: ', JSON.stringify(eosDidDocument, null, 2));
-  })
-
-  it('Resolve a jungle testnet DID Document', async () => {
-    const jungleDid = 'did:eosio:eos:testnet:jungle:lioninjungle';
-    console.log('Resolving ' + jungleDid);
-    const jungleDidDocument = await resolver.resolve(jungleDid, { fetch });
-    console.log('DID Document: ', JSON.stringify(jungleDidDocument, null, 2));
-  })
-
-  it('Resolve a chain-id based DID Document', async () => {
-    const telosDid = 'did:eosio:4667b205c6838ef70ff7988f6e8257e8be0e1284a2f59699054a018f743b1d11:caleosblocks';
-    console.log('Resolving ' + telosDid);
-    const telosDidDocument = await resolver.resolve(telosDid, { fetch });
-    console.log('DID Document: ', JSON.stringify(telosDidDocument, null, 2));
-  })
-
-  it('Fail to resolve an invalid DID', async () => {
-    const invalidDid = 'did:eosio:eos:aaaaaaaaaaaaaaaaaaaa';
-    console.log('Resolving ' + invalidDid);
-    try {
-      const invalidDidDocument = await resolver.resolve(invalidDid, { fetch });
-      console.log('DID Document:', invalidDidDocument);
-    } catch (e) {
-      console.log('Error resolving DID:', e);
-    }
-  })
-
-  it('Fail to resolve a non existant DID', async () => {
-    const nonExistantDid = 'did:eosio:telos:aaaaa';
-    console.log('Resolving ' + nonExistantDid);
-    try {
-      const nonExistantDidDocument = await resolver.resolve(nonExistantDid, { fetch });
-      console.log('DID Document:', nonExistantDidDocument);
-    } catch (e) {
-      console.log('Error resolving DID:', e);
-    }
-  })
-
+  const responses = testDids.map(({ did, expectedResult }) => ({
+    did,
+    expectedResult,
+    asyncResponse: resolver.resolve(did, { fetch }),
+  }));
+  describe('Dids resolve to DIDResolutionResult', () => {
```

Comment:
does this resolve each DID test several times (one time for each `describe`) or only once and the result is then inspected several times?

It should be the latter.

Reply:
Every did is resolved only once, the result is then tested multiple times. There are 4 tests for 7 dids, so 28 tests for the resolver. The resolver is called 7 times.